src/commands.cc

```diff
@@ -412,30 +412,42 @@ static void ensure_all_buffers_are_saved()
     throw runtime_error(message);
 }
 
+template<bool force>
+void kill(const ParametersParser& parser, Context& context, const ShellContext&)
+{
+    if (not force and ClientManager::instance().count() == 1)
+        ensure_all_buffers_are_saved();
+
+    const int status = parser.positional_count() > 0 ? str_to_int(parser[0]) : 0;
+    for (auto& client : ClientManager::instance())
```

Comment:
This will be buggy, as once we start removing clients, the iterators are going to be invalidated. This should be something like `while (not ClientManager::instance().empty()) { ... }`